client/python/openlineage/client/transport/composite.py

```diff
@@ -83,10 +83,18 @@ def emit(self, event: Event) -> None:
             try:
                 log.debug("Emitting event using transport %s", transport)
                 transport.emit(event)
-            except Exception as e:  # noqa: BLE001
+            except Exception as e:
                 if self.config.continue_on_failure:
                     log.warning("Transport %s failed to emit event with error: %s", transport, e)
                     log.debug("OpenLineage emission failure details:", exc_info=True)
                 else:
                     msg = f"Transport {transport} failed to emit event"
                     raise RuntimeError(msg) from e
+
+    def wait_for_completion(self, timeout: float = -1.0) -> bool:
+        # This can wait longer than timeout if multiple transports are slow, but acceptable
+        return all(transport.wait_for_completion(timeout) for transport in self.transports)
+
+    def close(self) -> None:
```

Comment:
```suggestion
    def shutdown(self, timeout: float = -1) -> bool:
        # This can wait longer than timeout if multiple transports are slow, but acceptable
        return all(transport.shutdown(timeout) for transport in self.transports)
        
    def wait_for_completion(self, timeout: float = -1) -> bool:
        return all(transport.wait_for_completion(timeout) for transport in self.transports)

    def close(self) -> None:
```
If some transport has custom `.shutdown()` method, it should be called.

Also TransformTransport should call `.shutdown()`, `.wait_for_completion()` and `.close()` of underlying transport as well.

Reply:
Also, both shutdown and close should be more reliable - if some of underlying transport fails to close, then `.shutdown()` will not be called for the rest of them:
```python
def shutdown(self, timeout: float = -1):
  result = True
  last_exception: Exception | None = None
  for transport in self.transports:
    try:
      result = transport.shutdown(timeout) and result
    except Exception as e:
      last_exception = e

  if last_exception:
    raise exception
  return result
```